### PR TITLE
Fix filename truncation #150

### PR DESCRIPTION
**Pull request by tarrencev**
Address #150 by specifying the filename (and parent id) in JSON instead of the Content-Disposition header

@gcurtis @codyebberson @dmaynard

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @tarrencev has signed the CLA. Thanks for the pull request!

File: src/main/java/com/box/sdk/BoxMultipartRequest.java [hunk 1]
```diff
@@ -120,7 +120,7 @@ protected void writeBody(HttpURLConnection connection, ProgressListener listener
                 this.writeOutput(entry.getValue());
             }
 
-            this.writePartHeader(new String[][] {{"name", "filename"}, {"filename", this.filename}},
+            this.writePartHeader(new String[][] {{"name", "file"}, {"filename", this.filename}},
```
Comment (codyebberson) [inline src/main/java/com/box/sdk/BoxMultipartRequest.java]: Is there a reason for "name=file" in the Content-Disposition header?  AFAIK, it's not part of the standard.  Maybe legacy support?  If no good reason, then perhaps we should remove it.

Also, it appears the writePartHeader still has escaping problems:
https://github.com/box/box-java-sdk/blob/master/src/main/java/com/box/sdk/BoxMultipartRequest.java#L177-L182

Line 181 in particular is directly writing user-submitted data to a HTTP header.  That could be abused.  We should URL-encode the value.

See:  http://stackoverflow.com/questions/93551/how-to-encode-the-filename-parameter-of-content-disposition-header-in-http/6745788#6745788

**Comment by codyebberson** [comment]
Minor nits:
* The encode method could be static, so you don't need to create a new QueryStringBuilder
* We could also remove the encode method entirely and use Java's built in [URLEncoder class](http://docs.oracle.com/javase/6/docs/api/java/net/URLEncoder.html)

Otherwise lgtm, +1

**Comment by tarrencev** [comment]
I switched it to using `URLEncode` earlier :) Makes sense to move `QueryStringBuilder` to that too but I'll leave that for another PR

**Pull request merged**